File: CPP/Kode Program/DPR.cpp
/*
Saya Muhammad Rizki Revandi dengan NIM 2205027 mengerjakan Latihan Praktikum 1
dalam mata kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya maka saya tidak
melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/

// library yang digunakan
#include <iostream>
#include <string>

// import standard namespace
using namespace std;

// deklarasi kelas DPR
class DPR
{
    // private attributes untuk id, nama, bidang, dan partai
    private:
        string id;
        string name;
        string bidang;
        string partai;
    
    // public attributes
    public:
        // constructor
        DPR()
        {
            // set id, nama, bidang, dan partai ke default
            this->id = "";
            this->name = "";
            this->bidang = "";
            this->partai = "";
        }

        DPR(string id, string name, string bidang, string partai)
        {
            this->id = id;
            this->name = name;
            this->bidang = bidang;
            this->partai = partai;
        }

        string get_id()
        {
            return this->id;
        }

        void set_id(string id)
        {
            this->id = id;
        }

        string get_name()
        {
            return this->name;
        }

        void set_name(string name)
        {
            this->name = name;
        }

        string get_bidang()
        {
            return this->bidang;
        }

        void set_bidang(string bidang)
        {
            this->bidang = bidang;
        }

        string get_partai()
        {
            return this->partai;
        }

        void set_partai(string partai)
        {
            this->partai = partai;
        }

        // destructor
        ~DPR()
        {
        }
};

File: PHP/DPR.php
<?php

// deklarasi kelas DPR
class DPR
{
    // atribut - atribut privat untuk id, nama, bidang, partai, dan foto
    private $id = "";
    private $nama = "";
    private $bidang = "";
    private $partai = "";
    private $foto = "";

    // konstruktor dengan parameter id, nama, bidang, partai, dan foto
    public function __construct($id = "", $nama = "", $bidang = "", $partai = "", $foto = "")
    {
        $this->id = $id;
        $this->nama = $nama;
        $this->bidang = $bidang;
        $this->partai = $partai;
        $this->foto = $foto;
    }

    // get id
    public function getId()
    {
        return $this->id;
    }

    // set id
    public function setId($id)
    {
        $this->id = $id;
    }

    // get nama
    public function getNama()
    {
        return $this->nama;
    }

    // set nama
    public function setNama($nama)
    {
        $this->nama = $nama;
    }

    // get bidang
    public function getBidang()
    {
        return $this->bidang;
    }

    // set bidang
    public function setBidang($bidang)
    {
        $this->bidang = $bidang;
    }

    // get partai
    public function getPartai()
    {
        return $this->partai;
    }

    // set partai
    public function setPartai($partai)
    {
        $this->partai = $partai;
    }

    // get foto
    public function getFoto()
    {
        return $this->foto;
    }

    // set foto
    public function setFoto($foto)
    {
        $this->foto = $foto;
    }
}

?>